loopy/statistics.py

```diff
@@ -170,51 +181,201 @@ def map_if_positive(self, expr):  # implemented in FlopCounter
+            return TypeToCountMap(
+                    {(self.type_inf(expr), 'nonconsecutive'): 1}
+                    ) + self.rec(expr.index)
+
+        # check coefficient of local_id0 for each axis
+        from loopy.symbolic import CoefficientCollector
+        from pymbolic.primitives import Variable
+        for idx, axis_tag in zip(index, array.dim_tags):
+
+            coeffs = CoefficientCollector()(idx)
+            # check if he contains the lid 0 guy
+            try:
+                coeff_id0 = coeffs[Variable(local_id0)]
+            except KeyError:
+                # does not contain local_id0
+                continue
+
+            if coeff_id0 != 1:
+                # non-consecutive access
+                return TypeToCountMap(
+                        {(self.type_inf(expr), 'nonconsecutive'): 1}
+                        ) + self.rec(expr.index)
+
+            # coefficient is 1, now determine if stride is 1
+            from loopy.kernel.array import FixedStrideArrayDimTag
+            if isinstance(axis_tag, FixedStrideArrayDimTag):
+                stride = axis_tag.stride
+            else:
+                continue
+
+            if stride != 1:
+                # non-consecutive
+                return TypeToCountMap(
+                        {(self.type_inf(expr), 'nonconsecutive'): 1}
+                        ) + self.rec(expr.index)
+
+            # else, stride == 1, continue since another idx could contain id0
+
+        # loop finished without returning, stride==1 for every instance of local_id0
+        return TypeToCountMap(
+                {(self.type_inf(expr), 'consecutive'): 1}
+                ) + self.rec(expr.index)
+
+    def map_sum(self, expr):
+        if expr.children:
+            return sum(self.rec(child) for child in expr.children)
+        else:
+            return TypeToCountMap()
+
+    map_product = map_sum
+
+    def map_quotient(self, expr, *args):
+        return self.rec(expr.numerator) + self.rec(expr.denominator)
+
+    map_floor_div = map_quotient
+    map_remainder = map_quotient
+
+    def map_power(self, expr):
+        return self.rec(expr.base) + self.rec(expr.exponent)
+
+    def map_left_shift(self, expr):
+        return self.rec(expr.shiftee)+self.rec(expr.shift)
+
+    map_right_shift = map_left_shift
+
+    def map_bitwise_not(self, expr):
+        return self.rec(expr.child)
+
+    def map_bitwise_or(self, expr):
+        return sum(self.rec(child) for child in expr.children)
+
+    map_bitwise_xor = map_bitwise_or
+    map_bitwise_and = map_bitwise_or
+
+    def map_comparison(self, expr):
+        return self.rec(expr.left)+self.rec(expr.right)
+
+    map_logical_not = map_bitwise_not
+    map_logical_or = map_bitwise_or
+    map_logical_and = map_logical_or
+
+    def map_if(self, expr):
+        warnings.warn("ExpressionSubscriptCounter counting DRAM accesses as "
+                      "sum of if-statement branches.")
```

Comment:
This warning is fine, since it points out something that's guaranteed to be a bit incorrect.